### ISSUE TITLE
Cannot Read Property "Initialize" of Null

### ISSUE THREAD

**Issue by csalazar1** [cypress-io/cypress#6537]
There is an issue in where cypress logs into the website, but there's a couple of uncaught errors that happen and the page fails to load. A handful of GET requests also return with property "canceled" as "true" 
<img width="1918" alt="Screenshot_1" src="https://user-images.githubusercontent.com/32462093/75099890-c56fd380-557b-11ea-9e7a-8404afab6707.png">
<img width="1913" alt="Screenshot_2" src="https://user-images.githubusercontent.com/32462093/75099892-c9035a80-557b-11ea-8b63-3a9ebd653b0b.png">

**Steps to Reproduce**
Simply run this code as a new test in Cypress **(you will be going into a sandbox with a test user)**
```js
describe('Sample Test', () => {
    it('Log into App', () => {
        cy.viewport(1920, 1080)
        cy.on('uncaught:exception', (err, runnable) => 
        {
            return false;
        })
        cy.visit('next-core.servicetitan.com/auth/newauth/login')
        cy.get('[placeholder*=\'Username\']').type('testuser7')
        cy.get('[placeholder*=\'Password\']').type('Password1@')
        cy.get('[type*="submit"]').click()
    })
  })
```

<img width="521" alt="Screenshot_3" src="https://user-images.githubusercontent.com/32462093/75099959-6e1e3300-557c-11ea-9061-257ea4731dc1.png">

The cypress.json file only contains 
```json
{
"chromeWebSecurity": false,
"pageLoadTimeout" : 120000
}
```
**Expected Results**
1. Log in successfully
2. Dashboard page loads properly

**Actual Results**
1. Logs in successfully
2. Dashboard page does **not** load and some errors are thrown

Cypress version = 4.0.2
Chrome browser version = 80

**Comment by jennifer-shehane** [comment]
Pending XHR requests are cancelled whenever your application navigates to a new URL. This matches the behavior of the browser - when you navigate to a new URL, the browser does not 'wait' for pending XHR requests from the previous tab/url to finish before moving on. 

You can see these cancelled requests are happening when a `NEW URL` event is logged. This is why. 

If your test is dependent on these XHR requests finishing before moving on to a new URL, we suggest writing an assertion using `cy.route()` and `cy.wait()` to wait for the XHR requests before moving on. 

We fail tests on uncaught errors thrown from your application or from one of your application's 3rd party plugins, you can ignore these errors as you have done using the `uncaught:exception` handler. We would like to show a warning on uncaught:exception instead of fail, but there's no work done on this yet https://github.com/cypress-io/cypress/issues/4917

**Comment by csalazar1** [comment]
@jennifer-shehane Thank you so much for the response. I just tried using cy.route() and cy.wait(), but they do not seem to be picking up some of the XHRs. 
<img width="780" alt="Screenshot_2" src="https://user-images.githubusercontent.com/32462093/75205369-71eeb880-5728-11ea-8f16-22fc40bde0e0.png">
Here's the code where I route **some** of the XHRs
<img width="412" alt="Screenshot_1" src="https://user-images.githubusercontent.com/32462093/75205418-921e7780-5728-11ea-941b-887167bd29bb.png">
And as you can see here, they're still getting canceled. 

The things I've also tried:
- Adding every single canceled XHR to the array, not just the ones that come before the URL change
- Waiting for each route individually one by one in order or execution
- Adding request timeout argument to cy.wait

Please advice

**Comment by jennifer-shehane** [comment]
Well, adding `cy.wait()` is not going to cancel the reroutes going through your new URL. 

What's happening here is when you click the button - your page reroutes to the following URLs in succession:

- https://go.servicetitan.com/
- https://go.servicetitan.com/#/Dashboard
- https://go.servicetitan.com/__/#/Dashboard

The pending XHR requests that started upon navigation are cancelled each time your application navigates to a new URL.

The reroute to https://go.servicetitan.com/__/#/Dashboard looks a bit strange though and may be related to this issue: https://github.com/cypress-io/cypress/issues/3975

**Comment by csalazar1** [comment]
So the fix needs to come from developers for the app itself, correct? there's no workaround for these new URL events?

**Comment by jennifer-shehane** [comment]
@csalazar1 There is no 'fix' necessary. XHR requests are cancelled when an application routes to a new URL. The Cypress command is simply showing this behavior.

**Comment by csalazar1** [comment]
@jennifer-shehane the original issue is that the application's dashboard does not load after logging in. It's like you said, this is due to the XHR requests getting cancelled by new URL events. So that was my question here, how can I get it to load if the requests get cancelled? Waiting for each request didn't work as you saw above

**Comment by jennifer-shehane** [comment]
If the issue is that the page does not load as expected, I do not think the aborted requests are related. It is a red herring. 

Please provide a reproducible example of your application's inability to load in Cypress and we can consider reopening.

**Comment by csalazar1** [comment]
@jennifer-shehane here's the code to repro the case
`describe('Sample Test', () => {
    it('Log into App', () => {
        cy.on('uncaught:exception', (err, runnable) => 
        {
            return false;
        })
        cy.viewport(1920, 1080)
        cy.visit('next-core.servicetitan.com/auth/newauth/login')
        cy.get('[placeholder*=\'Username\']').type('wolfadmin3')
        cy.get('[placeholder*=\'Password\']').type('Turok123$')
        cy.get('[type*="submit"]').click()
    })
  })`

This is expected dashboard load
<img width="1897" alt="Screenshot_1" src="https://user-images.githubusercontent.com/32462093/75691657-35a3e680-5c59-11ea-8670-ef5448fcbb43.png">

This is actual behavior on Cypress
<img width="1916" alt="Screenshot_2" src="https://user-images.githubusercontent.com/32462093/75691682-3dfc2180-5c59-11ea-9b8b-ab53abfdc176.png">

**Comment by jennifer-shehane** [comment]
@csalazar1 Yes, I can reproduce. As stated in my earlier comment, the reroute to go.servicetitan.com/__/#/Dashboard is related to this issue: #3975

Duplicate of #3975

**Comment by csalazar1** [comment]
Excellent! Thank you so much for all your time and effort Jennifer. It is greatly appreciated :)

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
